### PR TITLE
Sprint 55

### PR DESCRIPTION
**Pull request by ericsorenson**
- [7080 BUG: Sort order of cases on Inventory Report](https://github.com/flexion/ef-cms/issues/7080) 
- [7081 BUG: Names with hypens NOT returning in case search](https://github.com/flexion/ef-cms/issues/7081) 
- [7179 Public Entity Validation / Security issue](https://github.com/flexion/ef-cms/issues/7179) 
- [7184 BUG: Stricken document not displaying stricken correctly ](https://github.com/flexion/ef-cms/issues/7184) 
- [7221 BUG: eAccess flag not showing for spouse](https://github.com/flexion/ef-cms/issues/7221) 
- [6229 Pro-Se Petitioners: Clinic Information on Dashboard](https://github.com/flexion/ef-cms/issues/6229) 
- [7178 Docket Clerk: Ability to mark something as pending after it’s been processed](https://github.com/flexion/ef-cms/issues/7178) 
- [7197 Court:  Petitioners with the eAccess flag should have electronic service.](https://github.com/flexion/ef-cms/issues/7197) 
- [7198 Court: Pending Items w/ Legacy Served not showing in UI](https://github.com/flexion/ef-cms/issues/7198) 
- [7199 Court: Indicator on Sealed Documents](https://github.com/flexion/ef-cms/issues/7199) 
- [7226 Sealed in Blackstone docs -- Non-Viewable to Parties/Practitioners on Case](https://github.com/flexion/ef-cms/issues/7226)

### CODE REVIEW

File: shared/src/business/entities/cases/Case.js [hunk 1]
```diff
@@ -1903,8 +1886,20 @@ Case.prototype.hasPartyWithPaperService = function () {
   );
 };
 
+const isSealedCase = rawCase => {
```
Comment (julialeague) [inline shared/src/business/entities/cases/Case.js:1889]: Can we get a test for this?

File: shared/src/utilities/JoiValidationDecorator.js [hunk 2]
```diff
@@ -195,17 +195,11 @@ exports.joiValidationDecorator = function (
...
         entityConstructor.validationName,
-        JSON.stringify(
-          stringifyTransform(pick(this, helpfulKeys)),
-          (key, value) =>
-            this.hasOwnProperty(key) && typeof value === 'undefined'
-              ? '<undefined>'
-              : value,
-        ),
-        JSON.stringify(stringifyTransform(this.getValidationErrors())),
+        JSON.stringify(stringifyTransform(validationErrors)),
```
Comment (julialeague) [inline shared/src/utilities/JoiValidationDecorator.js:195]: is the `helpfulKeys` constant on line 180 needed anymore with this change?

File: web-api/migration-terraform/main/lambdas/migration-segments.js [hunk 3]
```diff
@@ -31,10 +38,17 @@ const dynamoDbDocumentClient = new AWS.DynamoDB.DocumentClient({
...
 const migrateRecords = async ({ documentClient, items }) => {
+  console.log('about to run migration 001');
   items = await migration0001(items, documentClient);
+  console.log('about to run migration 002');
   items = await migration0002(items, documentClient);
+  console.log('about to run migration 003');
   items = await migration0003(items, documentClient);
+  console.log('about to run migration 004');
   items = await migration0004(items, documentClient);
+  items = await migration0005(items, documentClient);
```
Comment (julialeague) [inline web-api/migration-terraform/main/lambdas/migration-segments.js:50]: Can we add a log message like the ones above for migration 005 and 006 as well?

File: web-api/migration-terraform/main/lambdas/migrations/0005-block-case-with-legacy-served-and-pending-items.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  Case,
+} = require('../../../../../shared/src/business/entities/cases/Case');
+const {
+  createISODateString,
+} = require('../../../../../shared/src/business/utilities/DateHandler');
+const { cloneDeep } = require('lodash');
+
+const applicationContext = createApplicationContext({});
+
+const migrateItems = async (items, documentClient) => {
```
Comment (julialeague) [inline web-api/migration-terraform/main/lambdas/migrations/0005-block-case-with-legacy-served-and-pending-items.js:61]: The nested if/elses along with the size of this code block make it hard to read - it would be nice to break these kinds of blocks out into functions.

File: web-api/migration-terraform/main/lambdas/migrations/0006-eservice-indicator-has-eaccess.test.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+    expect(results).toEqual([
+      expect.objectContaining({
+        ...caseRecords[0],
+        contactPrimary: expect.objectContaining({
+          serviceIndicator: SERVICE_INDICATOR_TYPES.SI_ELECTRONIC,
+        }),
+      }),
+      expect.objectContaining(caseRecords[1]),
+      expect.objectContaining(nonCaseRecords[0]),
```
Comment (julialeague) [inline web-api/migration-terraform/main/lambdas/migrations/0006-eservice-indicator-has-eaccess.test.js:87]: Can we check that the nonCaseRecords aren't altered here?

File: web-client/src/presenter/actions/actionFactory.js [hunk 6]
```diff
@@ -1,51 +1,6 @@
...
-
-import { setAddEditUserCaseNoteModalStateFromDetailAction as setAddEditUserCaseNoteModalStateFromDetailActionOld } from './TrialSessionWorkingCopy/setAddEditUserCaseNoteModalStateFromDetailAction.old';
-import { setJudgesCaseNoteOnCaseDetailAction as setJudgesCaseNoteOnCaseDetailActionOld } from './TrialSession/setJudgesCaseNoteOnCaseDetailAction.old';
-import { setViewerDocumentToDisplayAction as setViewerDocumentToDisplayActionOld } from './setViewerDocumentToDisplayAction.old';
-
-import { generateTitleAction } from './FileDocument/generateTitleAction';
-import { generateTitleForPaperFilingAction } from './FileDocument/generateTitleForPaperFilingAction';
-
-import { isCodeDisabled } from '../../../../codeToggles';
+// import { isCodeDisabled } from '../../../../codeToggles';
```
Comment (julialeague) [inline web-client/src/presenter/actions/actionFactory.js]: Could this file be deleted?

File: web-client/src/presenter/actions/getCaseAssociationAction.test.js [hunk 7]
```diff
@@ -155,7 +155,9 @@ describe('getCaseAssociation', () => {
       props: {},
       state: {
         caseDetail: {
-          userId: '123',
+          contactPrimary: {
```
Comment (julialeague) [inline web-client/src/presenter/actions/getCaseAssociationAction.test.js:158]: Should the test case below this one be updated similarly on line 186, to have a contactId in the caseDetail instead of a userId?

**Comment by julialeague** [comment]
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [x] If changes were made in the UI (if not, skip):
  - [x] If desired, run `docker system prune` to remove any unused docker images from previous code review.
  - [x] Fetch the pull request for the sprint (e.g., `git fetch origin pull/{PR #}/head:sprint-{Sprint #}`), and then switch to that branch (e.g. `git checkout sprint-{Sprint #}`)
  - [x] stand up the site locally, with `./docker-run.sh`
	- [x] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
	- [x] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
	- [x] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/)
	- [x] test in Mobile Safari and Mobile Chrome (or an emulator like Chrome DevTools), with the caveat that not all internal Court functionality will be necessary on these platforms
	- [x] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [x] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [x] review for accessibility
		- [x] use an automated audit tool, such as Chrome Audit or aXe
		- [x] navigate site only with the keyboard
		- [x] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [x] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [x] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [x] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [x] review all tests
	- [x] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [x] methodically review all new tests for correctness, quality of naming
- [x] determine what code isn’t tested, review that rigorously
- [x] review documentation to ensure that it matches changes
	- [x] user-visible changes (including API users like the IRS) are documented in CHANGELOG.md (which is linked from US Tax Court website).
- [x] provide comments on the pull request on GitHub, as necessary
	- [x] for comments that are specific to a particular line of code, comment on those specific lines
- [x] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: web-client/src/presenter/actions/actionFactory.js (hunk 6 above)
Comment (kkoskelin) [reply web-client/src/presenter/actions/actionFactory.js]: I'm removing the commented-out import.  However, we're going to leave the `codeToggles.js` file in place because it's a piece of code that may be used in the near-future for larger feature-work.

File: web-client/src/presenter/actions/getCaseAssociationAction.test.js (hunk 7 above)
Comment (kkoskelin) [reply web-client/src/presenter/actions/getCaseAssociationAction.test.js:158]: Modifications made in relevant test at 186.

**Review by julialeague — APPROVED** [review]
@mmarcotte, I recommend that you accept these changes into the court's environments! 🙇‍♀️
